src/Fleck/Handlers/Hybi13Handler.cs

```diff
@@ -47,7 +48,36 @@ public static byte[] FrameData(byte[] payload, FrameType frameType)
             
             return memoryStream.ToArray();
         }
-        
+
+        public static byte[] FrameData(ArraySegment<byte> payload, FrameType frameType)
+        {
+            var memoryStream = new MemoryStream();
+            byte op = (byte)((byte)frameType + 128);
+
+            memoryStream.WriteByte(op);
+
+            if (payload.Count > UInt16.MaxValue)
+            {
+                memoryStream.WriteByte(127);
+                var lengthBytes = payload.Count.ToBigEndianBytes<ulong>();
+                memoryStream.Write(lengthBytes, 0, lengthBytes.Length);
+            }
+            else if (payload.Count > 125)
+            {
+                memoryStream.WriteByte(126);
+                var lengthBytes = payload.Count.ToBigEndianBytes<ushort>();
+                memoryStream.Write(lengthBytes, 0, lengthBytes.Length);
+            }
+            else
+            {
+                memoryStream.WriteByte((byte)payload.Count);
+            }
+
+            memoryStream.Write(payload.Array, 0, payload.Count);

```

Comment:
Should `0` be `payload.Offset` so it uses the correct range from the Segment's array?

Reply:
Yep, fixed that